Sprint-1/implement/sum.test.js

```diff
@@ -13,24 +13,43 @@ const sum = require("./sum.js");
 // Given an empty array
 // When passed to the sum function
 // Then it should return 0
-test.todo("given an empty array, returns 0")
+test("Given an empty array,  it should return 0", () => {
+  expect(sum([])).toEqual(0);
+});
 
 // Given an array with just one number
 // When passed to the sum function
 // Then it should return that number
+test("Given an array with just one number,  it should return that number", () => {
+  expect(sum([666])).toEqual(666);
+});
 
 // Given an array containing negative numbers
 // When passed to the sum function
 // Then it should still return the correct total sum
+test("Given an array containing negative numbers,  it return the correct total sum", () => {
+  expect(sum([-2, -5, -7, -44])).toEqual(-58);
+});
+
 
 // Given an array with decimal/float numbers
 // When passed to the sum function
 // Then it should return the correct total sum
+test("Given an array with decimal/float numbers,  it return the correct total sum", () => {
+  expect(sum([2.04, 5.3, 7.5, 4.4])).toEqual(19.240000000000002);
```

Comment:
Decimal numbers in most programming languages (including JS) are internally represented in "floating point number" format. Floating point arithmetic is not exact. For example, the result of `46.5678 - 46 === 0.5678` is false because `46.5678 - 46` only yield a value that is very close to `0.5678`.

So that is why `expect(sum([2.04, 5.3, 7.5, 4.4])).toEqual(19.24);` fails.

`expect(sum([2.04, 5.3, 7.5, 4.4])).toEqual(19.240000000000002);` may not always work because on a different implementation, the resulting sum can be different if the numbers are added in different order.
```
2.04 + 5.3 + 7.5 + 4.4 == 19.240000000000002   // true

4.4 + 5.3 + 2.04 + 7.5 == 19.240000000000002   // false
```

Can you find a more appropriate way to test a value (that involves decimal number calculations) for "equality?

Reply:
Yes, there are ways to handle this issue: one approach is using expect(Math.abs(result - expected) < epsilon).toBe(true) to compare with a small tolerance, and another is leveraging a library like Math.js and testing with expect(result).toBeCloseTo(expected, 10) for precision.